src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -29,11 +30,14 @@ public sealed partial class JsonSerializerOptions
         [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
         private static void RootReflectionSerializerDependencies()
         {
-            if (s_defaultSimpleConverters is null)
+            // s_typeInfoCreationFunc is the last field assigned.
+            // Use it as the sentinel to ensure that all dependencies are initialized.
+            if (s_typeInfoCreationFunc is null)
```

Comment:
Thought about his a little bit more, and you may want to use `Volatile.Read(ref ...)` here as well. The reason is that we don't want an optimizing compiler / CPU to attempt to read *s_defaultSimpleConverters* or *s_defaultFactoryConverters* before *s_typeInfoCreationFunc* is confirmed not-null. Using `Volatile.Read` here to complement the `Volatile.Write` you have below will help guarantee this can never happen on the read side.

Reply:
Hmm, we might want to service 6.0 with this change since that particular change was cherry picked in #65898.